### PR TITLE
Log line number fix and added better contextual support for PyQt5

### PR DESCRIPTION
**Pull request by nicoddemus**
@The-Compiler, what do you think? It will be a little more verbose when using PyQt5, but I think the extra information can be useful.

### CODE REVIEW

**Comment by The-Compiler** [comment]
Yeah, the information definitely *is* useful!

What about adding a newline and indenting the line after the source information?

Currently:

```
/home/florian/proj/qutebrowser/git/tests/utils/test_log.py:test_unfiltered:237: QtWarningMsg: Hello World
/home/florian/proj/qutebrowser/git/tests/utils/test_log.py:test_unfiltered:238: QtWarningMsg: Another one
```

Proposed:

```
/home/florian/proj/qutebrowser/git/tests/utils/test_log.py:test_unfiltered:237:
    QtWarningMsg: Hello World
/home/florian/proj/qutebrowser/git/tests/utils/test_log.py:test_unfiltered:238:
    QtWarningMsg: Another one
```

File: pytestqt/plugin.py [hunk 1]
```diff
@@ -709,7 +721,14 @@ def _handle(self, msg_type, message):
                 ignored = True
                 break
 
-        self._records.append(Record(msg_type, message, ignored))
+        if context is not None:
+            context = self._Context(
+                to_unicode(context.file),
+                to_unicode(context.function),
+                context.line,
```
Comment (The-Compiler) [inline pytestqt/plugin.py]: Note `context.file` and `context.function` can also be `None` in rare circumstances. I don't really know when it happens, but I have seen it happen (and handle it in my custom message handler).

**Comment by nicoddemus** [comment]
Thanks for the suggestions, I will apply them tomorrow. :smile:

About `context.file` and `context.function` being `None`, I guess in those circumstances the output will contain `None` in those fields... not sure if there's much more that can be done since I opted to use string interpolation to configure formatting and it doesn't support conditionals.

**Comment by The-Compiler** [comment]
> About context.file and context.function being None, I guess in those circumstances the output will contain None in those fields... not sure if there's much more that can be done since I opted to use string interpolation to configure formatting and it doesn't support conditionals.

That sounds good enough to me - maybe there should be a test for that scenario to make sure nothing raises `TypeError` or so because of the `None`?

**Comment by nicoddemus** [comment]
> That sounds good enough to me - maybe there should be a test for that scenario to make sure nothing raises TypeError or so because of the None?

Sure, good idea.

**Comment by nicoddemus** [comment]
Thanks for the review @The-Compiler! :smile:

**Pull request merged**